mergify_cli/__init__.py

```diff
@@ -437,36 +437,37 @@ async def log_httpx_response(response: httpx.Response) -> None:  # noqa: RUF029
     )
 
 
-def get_trunk() -> str:
+def get_branch() -> str:
     try:
-        trunk = subprocess.check_output(
-            "git config --get mergify-cli.stack-trunk",
-            shell=True,
+        return subprocess.check_output(
+            ["git", "rev-parse", "--abbrev-ref", "HEAD"],
             text=True,
         ).strip()
-    except subprocess.CalledProcessError:
-        trunk = ""
+    except subprocess.CalledProcessError as e:
+        console.print("error: can't get the current branch", style="red")
+        console.print(e.output, e.stderr)
+        raise
 
-    if not trunk:
-        try:
-            dest_branch = subprocess.check_output(
-                "git rev-parse --abbrev-ref HEAD",
-                shell=True,
-                text=True,
-            ).strip()
-        except subprocess.CalledProcessError:
-            return ""
 
-        try:
-            trunk = subprocess.check_output(
-                f"git for-each-ref --format='%(upstream:short)' refs/heads/{dest_branch}",
-                shell=True,
-                text=True,
-            ).strip()
-        except subprocess.CalledProcessError:
-            trunk = ""
+def get_trunk() -> str:
+    branch = get_branch()
+    try:
+        target_branch = subprocess.check_output(
+            ["git", "config", "--get", "branch." + branch + ".merge"],
+            text=True,
+        ).strip()
+    except subprocess.CalledProcessError:
+        # It's possible this has not been set; ignore
+        console.print("error: can't get the remote target branch", style="red")
+        console.print(
```

Comment:
The high-level command is `git branch --set-upstream-to=origin/main`.

`git config --add` may not override existing key. rebase may later fail because the branch does not exist on the configured target remote.
Using `git branch --set-upstream-to=` ensure the config is valid and force user to also provide the right remote.